src/products/index.ts

```diff
@@ -0,0 +1,106 @@
+import { ApiRequest, TVersion, Version } from ".."
+import {
+  Plan,
+  Product,
+  ProductCapability,
+  ProductCurrency,
+  ProductPricingTable,
+} from '../types';
+import { v2ProductMethods } from "./v2";
+
+
+export type ProductVersions = {
+  [Version.V2]: {
+    /**
+     *  Retrieves a list of all products
+     *
+     * Docs - https://docs.salable.app/api/v2#tag/Products/operation/getProducts
+     *
+     * @returns {Promise<Product[]>} All products present on the account
+     */
+    getAll: () => Promise<Product[]>;
+
+    /**
+     *  Retrieves a specific product by its UUID. By default, the response does not contain any relational data. If you want to expand the relational data, you can do so with the `expand` query parameter.
+     *  @param {string} productUuid - The UUID of the product
+     *
```

Comment:
List options as a param with some generic description. Mention query parameters or body depending on what the options are and say refer to docs for available options.